cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -116,6 +116,11 @@ func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
 	return files, nil
 }
 
+// NewGitHubRepository returns a gitHubRepository implementation.
+func NewGitHubRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient, opts ...githubRepositoryOption) (Repository, error) {
```

Comment:
Q: Can we avoid this wrapper and just make the existing constructor public?

Reply:
Just tried this and I get this lint error
```
cmd/clusterctl/client/repository/repository_github.go:120:137: unexported-return: exported func NewGitHubRepository returns unexported type *repository.gitHubRepository, which can be annoying to use (revive)
func NewGitHubRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient, opts ...githubRepositoryOption) (*gitHubRepository, error) {

```

Reply:
so there were not many uses of the private struct, so i have changed the return to "Repository" and cast the usages in the unit tests..